crates/ibc/src/core/ics04_channel/handler/timeout.rs

```diff
@@ -133,9 +134,21 @@ pub fn process<Ctx: ChannelReader>(
 
     output.log("success: packet timeout ");
 
-    output.emit(IbcEvent::TimeoutPacket(TimeoutPacket {
-        packet: packet.clone(),
-    }));
+    output.emit(IbcEvent::TimeoutPacket(TimeoutPacket::new(
+        packet.clone(),
+        source_channel_end.ordering,
+    )));
+
+    if source_channel_end.order_matches(&Order::Ordered) {
```

Comment:
I didn't add the check since the channel [will always be closed](https://github.com/cosmos/ibc-rs/blob/af0b5372831ba350cc9b92bd973fccb9747f6b74/crates/ibc/src/core/ics04_channel/handler/timeout.rs#L111) when the channel is ordered, and it's a violation of the spec not to.

Reply:
Good point! I missed that! :+1: